application/ui/src/features/inspect/dataset/dataset-item/dataset-item.component.tsx

```diff
@@ -21,19 +23,42 @@ const DatasetItemPlaceholder = () => {
 };
 
 interface DatasetItemProps {
-    mediaItem: MediaItem | undefined;
+    mediaItem: MediaItem;
 }
 
-export const DatasetItem = ({ mediaItem }: DatasetItemProps) => {
-    if (mediaItem === undefined) {
-        return <DatasetItemPlaceholder />;
-    }
+const DatasetItem = ({ mediaItem }: DatasetItemProps) => {
+    const { selectedMediaItem, onSetSelectedMediaItem } = useSelectedMediaItem();
+    const { onInference, selectedModelId } = useInference();
 
-    const mediaUrl = `/api/projects/${mediaItem.project_id}/images/${mediaItem.id}/thumbnail`;
+    const isSelected = selectedMediaItem?.id === mediaItem.id;
+
+    const mediaUrl = `/api/projects/${mediaItem.project_id}/images/${mediaItem.id}/full`;
+
+    const handleClick = async () => {
+        onSetSelectedMediaItem(mediaItem);
+        selectedModelId !== undefined && (await onInference(mediaItem, selectedModelId));
+    };
 
     return (
-        <View UNSAFE_className={styles.datasetItem}>
+        <div
```

Comment:
One thing we can look into in the future is to see if we can reuse the dataset grid that Camilo made for Tune. This also supports virtualized lists by leveraging spectrum's virtualizer + some extra accessibility via their `ListBox` and `ListBoxItem` components.

You can see that related code here: https://github.com/open-edge-platform/training_extensions/blob/develop/application/ui/src/features/dataset/gallery/gallery.component.tsx